src/main/ruby/core/exception.rb

```diff
@@ -390,37 +401,56 @@ class KeyError < IndexError
 
 class SignalException < Exception
 
+  alias :signm :message
   attr_reader :signo
-  attr_reader :signm
-
-  def initialize(signo = nil, signm = nil)
-    # MRI overrides this behavior just for SignalException itself
-    # but not for anything that inherits from it, therefore we
-    # need this ugly check to make sure it works as intended.
-    return super(signo) unless self.class == SignalException
-    if signo.is_a? Integer
-      unless @signm = Signal::Numbers[signo]
-        raise ArgumentError, "invalid signal number #{signo}"
+
+  def initialize(*args)
```

Comment:
Can we use a simpler signature here like `sig, message=nil`?

Reply:
I picked this signature for compatibility.

Reply:
What compatibility?
The value of `arity`? Is that spec'd? I'd consider it an unimportant implementation detail of MRI in general.

Reply:
(MRI cannot handle optional arguments without changing the arity to -1 in methods defined in C iirc)

Reply:
@eregon Updated, I used `initialize(sig, message = undefined)` because the behavior appears to vary by the number of arguments.
```
irb(main):003:0> SignalException.new(2).signm
=> "SIGINT"
irb(main):004:0> SignalException.new(2, nil).signm
```